packages/frontend/src/components/Explorer/ExploreTree/TableTree/ItemDetailPreview.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+    Anchor,
+    Box,
+    Group,
+    Popover,
+    Text,
+    Title,
+    useMantineTheme,
+} from '@mantine/core';
+import { IconTable } from '@tabler/icons-react';
+import ReactMarkdownPreview from '@uiw/react-markdown-preview';
+import { FC, PropsWithChildren } from 'react';
+import { rehypeRemoveHeaderLinks } from '../../../../utils/markdownUtils';
+import MantineIcon from '../../../common/MantineIcon';
+import { useItemDetail } from './ItemDetailContext';
+
+/**
+ * Renders markdown for an item's description, with additional constraints
+ * to avoid markdown styling from completely throwing its surroundings out
+ * of whack.
+ */
+export const ItemDetailMarkdown: FC<{ source: string }> = ({ source }) => {
+    const theme = useMantineTheme();
+    return (
+        <ReactMarkdownPreview
+            skipHtml
+            linkTarget="_blank"
+            components={{
+                h1: ({ children }) => <Title order={2}>{children}</Title>,
+                h2: ({ children }) => <Title order={3}>{children}</Title>,
+                h3: ({ children }) => <Title order={4}>{children}</Title>,
+            }}
+            rehypeRewrite={rehypeRemoveHeaderLinks}
+            source={source}
+            disallowedElements={['img']}
+            style={{
+                fontSize: theme.fontSizes.sm,
+            }}
+        />
+    );
+};
+
+/**
+ * Renders a truncated version of an item's description, with an option
+ * to read the full description if necessary.
+ */
+export const ItemDetailPreview: FC<{
+    description?: string;
+    onViewDescription: () => void;
+}> = ({ description, onViewDescription }) => {
+    if (!description) return null;
+
+    /**
+     * This value is pretty arbitrary - it's an amount of characters that will exceed
+     * a single line, and for which the 'Read more' option should make sense, and not
+     * be an annoyance.
+     *
+     * It's better to err on the side of caution, and show the 'Read more' option even
+     * if unnecessarily so.
+     */
+    const isTruncated = description.length > 180;
+
+    return (
+        <Box>
+            <Box
+                mah={120}
+                style={{
+                    overflow: 'hidden',
+
+                    // If we're over the truncation limit, use a mask to fade out the bottom of the container.
+                    maskImage: isTruncated
+                        ? 'linear-gradient(180deg, white 0%, white 80%, transparent 100%)'
+                        : undefined,
```

Comment:
oh nice! I've never used these

Reply:
Yeah same! I had the idea in the back of my head it was a viable solution, but had never actually used it.

Forgot to mention re: using a mantine helper for gradients: ended up not using them because they are not very flexible in terms of how the gradient steps are spaced out!

Reply:
Agreed! It can be useful only if you're using Mantine colors and you don't want to hardcode a HEX. Otherwise, this approach is more readable.